backend/layers/api/portal_api.py

```diff
@@ -195,12 +195,24 @@ def _dataset_to_response(dataset: DatasetVersion, is_tombstoned: bool, is_in_pub
 
 
 def _collection_to_response(collection: CollectionVersionWithDatasets, access_type: str):
-    collection_id = collection.collection_id.id if collection.published_at is not None else collection.version_id.id
-
+    """
+    Converts a CollectionVersion to a format that can be used as an API response. The returned id
+    """
     if collection.canonical_collection.originally_published_at is not None and collection.published_at is None:
+        # In this case, the collection version is a revision of an already published collection.
+        # We should expose version_id as the collection_id
         revision_of = collection.collection_id.id
+        collection_id = collection.version_id.id
+    elif collection.canonical_collection.originally_published_at is None and collection.published_at is None:
+        # In this case, we're dealing with a freshly created collection - we should expose the canonical id here,
+        # since the curators will circulate the permalink immediately. `revision_of` is also null.
+        # Note that this case is effectively equivalent to a published collection
```

Comment:
nit: we could just put this in an `else` statement and keep the comment for clarity. But both should work

Reply:
We will need this same logic for the curation API. It might make sense to extract this logic to a shared location.

As for the above comments, I have no strong opinion either way.